### PR TITLE
chore(test): add test for error during redirect

### PR DESCRIPTION
**Pull request by tripodsan**
adding test for errors during redirect

### CODE REVIEW

File: test/fetch/index.test.js [hunk 1]
```diff
@@ -367,6 +368,32 @@ testParams.forEach((params) => {
       assert.strictEqual(resp.redirected, true);
     });
 
+    it('connection error in redirected location is handled correctly', async () => {
+      let server;
+      await new Promise((resolve, reject) => {
+        server = https.createServer({ }, (req, res) => {
```
Comment (tripodsan) [inline test/fetch/index.test.js:374]: since server doesn't have a valid cert, the connection is refused.

File: test/fetch/index.test.js [hunk 2]
```diff
@@ -367,6 +368,32 @@ testParams.forEach((params) => {
...
+
+      try {
+        const location = `${protocol}://localhost:${server.address().port}/`;
+        const url = `${protocol}://httpbingo.org/redirect-to?url=${encodeURIComponent(location)}&status_code=302`;
+        await fetch(url, { cache: 'no-store' });
+        assert.fail('redirect should fail');
+      } catch (e) {
+        if (e.message === 'redirect should fail') {
+          throw e;
+        }
```
Comment (tripodsan) [inline test/fetch/index.test.js:391]: here we could log the error....

**Comment by tripodsan** [comment]
@stefan-guggisberg shall I merge in this test?

**Comment by stefan-guggisberg** [comment]
> @stefan-guggisberg shall I merge in this test?

no, I’d like to improve it but didn’t get around to do it yet.

**Pull request merged**